### PR TITLE
fix: overload dump_write_request shouldn't auto-reply to client

### PR DESCRIPTION
**Pull request by Shuo-Jia**
### What problem does this PR solve? 
#533 overload `dump_write_request` for https://github.com/XiaoMi/rdsn/pull/414 and cause the problem that the server will auto-reply to client with `ERR_OK`, so all the `response_client_write` after `dump_write_request` will be ignored for client, it also mean client will always receive `ERR_OK`, though pass `ERR_INVALID_DATA` in https://github.com/XiaoMi/rdsn/pull/414.

### What is changed and how does it work?
This PR use rpc_holder constructor replace `rpc::auto_replay`to dump request

###  Checklist 
- Manual test (add detailed scripts or steps below)

##### Side effects
- large write request will be responsed with `ERR_INVALID_DATA` and throw exception at client side

##### Related changes

- Need to cherry-pick to the release branch
- Need to update the documentation
- Need to be included in the release note

### CODE REVIEW

**Review by levy5307 — DISMISSED** [review]

File: src/server/pegasus_server_impl.cpp [hunk 1]
```diff
@@ -2846,49 +2846,35 @@ std::string pegasus_server_impl::dump_write_request(dsn::message_ex *request)
...
-        std::string request("multi_put:");
-        request.append("hash_key=")
-            .append(pegasus::utils::c_escape_string((multi_put.hash_key))
-                        .append(",multi_put_count=")
-                        .append(std::to_string(multi_put.kvs.size())));
-        return request;
+        auto multi_put = multi_put_rpc(request).request();
+        return fmt::format("multi_put: hash_key={}, multi_put_count={}",
+                           pegasus::utils::c_escape_string(multi_put.hash_key),
+                           std::to_string(multi_put.kvs.size()));
```
Comment (Smityz) [inline src/server/pegasus_server_impl.cpp:2861]: no need to use `std::to_string`

File: src/server/pegasus_server_impl.cpp [hunk 2]
```diff
@@ -2846,49 +2846,35 @@ std::string pegasus_server_impl::dump_write_request(dsn::message_ex *request)
...
-            .append(",check_sort_key=")
-            .append(pegasus::utils::c_escape_string(check_and_mutate.check_sort_key))
-            .append(",set_value_count=")
-            .append(std::to_string(check_and_mutate.mutate_list.size()));
-        return request;
+        auto check_and_mutate = check_and_mutate_rpc(request).request();
+        return fmt::format("check_and_mutate: hash_key={}, check_sort_key={}, set_value_count={}",
+                           pegasus::utils::c_escape_string(check_and_mutate.hash_key),
+                           pegasus::utils::c_escape_string(check_and_mutate.check_sort_key),
+                           std::to_string(check_and_mutate.mutate_list.size()));
```
Comment (Smityz) [inline src/server/pegasus_server_impl.cpp:2877]: no need to use `std::to_string`

**Review by levy5307 — APPROVED** [review]

**Comment by neverchanje** [comment]
Good job. Is there any case of bug caused by this problem?

**Comment by Shuo-Jia** [comment]
> Good job. Is there any case of bug caused by this problem?

`dump_request_write` is used for `large size limit`, I haven't found other case

**Review by neverchanje — APPROVED** [review]

**Pull request merged**